tensorflow/lite/micro/kernels/split_v.cc

```diff
@@ -0,0 +1,168 @@
+
+  int64_t split_size = 0;
+  for (int i = 0; i < outputs_count; i++) {
+    split_size += GetOutput(context, node, i)->dims->data[axis];
+  }
+
+  TFLITE_DCHECK_EQ(split_size, input_dims->data[axis]);
+  int64_t outer_size = 1;
+  for (int i = 0; i < axis; ++i) {
+    outer_size *= input_dims->data[i];
+  }
+
+  int64_t base_inner_size = 1;
+  for (int i = axis + 1; i < split_dimensions; ++i) {
+    base_inner_size *= input_dims->data[i];
+  }
+
+  const T* input_ptr = GetTensorData<T>(input);
+  for (int k = 0; k < outer_size; ++k) {
+    for (int i = 0; i < outputs_count; ++i) {
+      TfLiteTensor* t = GetOutput(context, node, i);
+      T* output_data = GetTensorData<T>(t);
+      output_dims = t->dims;
+      const int copy_size = output_dims->data[axis] * base_inner_size;
+      T* output_ptr = output_data + k * copy_size;
+      for (int j = 0; j < copy_size; ++j) output_ptr[j] = input_ptr[j];
+      input_ptr += copy_size;
+    }
+  }
+
+  return kTfLiteOk;
+}
+
+TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
+  TF_LITE_ENSURE_EQ(context, NumInputs(node), 3);
+
+  OpContext op_context(context, node);
+
+  TF_LITE_ENSURE_EQ(context, NumOutputs(node), op_context.params->num_splits);
+
+  auto input_type = op_context.input->type;
+  TF_LITE_ENSURE(context,
+                 input_type == kTfLiteFloat32 || input_type == kTfLiteUInt8 ||
+                     input_type == kTfLiteInt16 || input_type == kTfLiteInt32 ||
+                     input_type == kTfLiteInt64 || input_type == kTfLiteInt8);
+  for (int i = 0; i < NumOutputs(node); ++i) {
+    GetOutput(context, node, i)->type = input_type;
+  }
+
+  auto size_splits = op_context.size_splits;
+  TF_LITE_ENSURE_EQ(context, NumDimensions(size_splits), 1);
+  TF_LITE_ENSURE_EQ(context, NumOutputs(node), NumElements(size_splits));
+
+  return kTfLiteOk;
+}
+
+TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
+  OpContext op_context(context, node);
+
+  // if not const return error - unsupported
+  if (!IsConstantTensor(op_context.axis) ||
+      !IsConstantTensor(op_context.size_splits)) {
+    context->ReportError(
+        context, "Only const tensors are supported for size_splits and axis");
+    return kTfLiteError;
+  }
+
+  switch (op_context.input->type) {
+    case kTfLiteFloat32: {
+      SplitImpl<float>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteUInt8: {
+      SplitImpl<uint8_t>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteInt16: {
+      SplitImpl<int16_t>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteInt32: {
+      SplitImpl<int32_t>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteInt64: {
+      SplitImpl<int64_t>(context, node, &op_context);
+	   break;
+    }
+    case kTfLiteInt8: {
+      SplitImpl<int8_t>(context, node, &op_context);
+
+      break;
+    }
+    default:
+      context->ReportError(context, "Type %s currently not supported.",

```

Comment:
`TF_LITE_KERNEL_LOG should be used instead of context->ReportError, so log strings can be removed to save space if needed.`

Reply:
Done